### ISSUE TITLE
"Cannot access a disposed object" when using an URL as PartitionKey

### ISSUE THREAD

**Issue by matthijsbreemans** [Azure/azure-storage-net#320]
When using 7.1.3-Preview on dotnet core 1.0.0  i'm getting the following exeception when using an URL (https://example.org/test/test/test/test/()test.html) as PartitionKey

```
       Message=Cannot access a disposed object.
Object name: 'System.Net.Http.WinHttpResponseStream'.
       Source=Microsoft.WindowsAzure.Storage
       StackTrace:
            at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.<ExecuteAsyncInternal>d__4`1.MoveNext() in C:\Program Files (x86)\Jenkins\workspace\release_dotnet_master\Lib\WindowsRuntime\Core\Executor\Executor.cs:line 315
         --- End of stack trace from previous location where exception was thrown ---
            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
            at xxx.ItemService.<Insert>d__3.MoveNext() in C:\Users\Matthijs\Documents\Projects\xxx\Services\ItemsService.cs:line 46
       InnerException: 
            HResult=-2146232798
            Message=Cannot access a disposed object.
Object name: 'System.Net.Http.WinHttpResponseStream'.
            ObjectName=System.Net.Http.WinHttpResponseStream
            Source=System.Net.Http
            StackTrace:
                 at System.Net.Http.WinHttpResponseStream.CheckDisposed()
                 at System.Net.Http.WinHttpResponseStream.ReadAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken token)
                 at System.Net.Http.DelegatingStream.ReadAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
                 at Microsoft.WindowsAzure.Storage.Core.Util.StreamExtensions.<WriteToAsync>d__1`1.MoveNext() in C:\Program Files (x86)\Jenkins\workspace\release_dotnet_master\Lib\Common\Core\Util\StreamExtensions.cs:line 299
              --- End of stack trace from previous location where exception was thrown ---
                 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
                 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                 at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.<ExecuteAsyncInternal>d__4`1.MoveNext() in C:\Program Files (x86)\Jenkins\workspace\release_dotnet_master\Lib\WindowsRuntime\Core\Executor\Executor.cs:line 179

```

But when I use a Guid as PartitionKey everything works fine.

Insert code:
```
CloudTableClient tableClient = storageAccount.CreateCloudTableClient();

            // Retrieve a reference to the table.
            CloudTable table = tableClient.GetTableReference("links");
            TableOperation insertOperation = TableOperation.Insert(new LinkItem(url, name)
            {
                SiteID = siteID,
                LinkID = linkID
            });
            
            // Execute the insert operation.
            return await table.ExecuteAsync(insertOperation);

  public class LinkItem : TableEntity
    {
     
        public string Name { get; set; }

        public string URL { get; set; }

        public int SiteID { get; set; }

        public string LinkID { get; set; }
        public LinkItem() { }
        public LinkItem(string url, string name)
        {
            this.PartitionKey = Guid.NewGuid().ToString();
            this.RowKey = name;
             
        }
    }
```

**Comment by asorrin-msft** [comment]
As far as I can tell, there are two issues here.  First, forward slashes aren't permitted in a partition key.  This is documented here:
https://msdn.microsoft.com/en-us/library/azure/dd179338.aspx
The second issue is that while this request should be failing on the service, something else has gone wrong in our error-handling.  The storage client should be throwing the HTTP error, not trying to access a disposed object (of course.)  We'll look into this and get back to you.

**Comment by GuardRex** [comment]
I have the same result (i.e., "Cannot access a disposed object.") when I accidentally didn't put quotes around a string value in a query string. For example when I screwed up and did this ...
```c#
query = $"PartitionKey eq 'EX' and RowKey ge {timestamp1} and RowKey le {timestamp2}";
```
Fixing the query worked, but as you, @asorrin-msft, said, the error handling failed on the malformed query.

**Comment by pemari-msft** [comment]
Fixed in 7.2.0 -- thanks for reporting this!
